lib/common/multithread.c

```diff
@@ -87,11 +87,16 @@ static void init_async(h2o_multithread_queue_t *queue, h2o_loop_t *loop)
 {
     int fds[2];
 
+#ifndef _WIN32
     if (cloexec_pipe(fds) != 0) {
         perror("pipe");
         abort();
     }
     fcntl(fds[1], F_SETFL, O_NONBLOCK);
+#else
+    u_long nonblock = 1;
```

Comment:
If you look a few lines above this is only relevant for the non libuv implementation
#if H2O_USE_LIBUV
#else

so this can not be the cause.

In cloexec to setup fds[0] and fds[1] on windows we have to create the socket before hand
This means we need to know if it is tcp or udp a stream etc beforehand which I don't think we know at this point ?